packages/spruce-next-helpers/src/skillskit/next/_document.js

```diff
@@ -41,6 +43,12 @@ export default class MyDocument extends Document {
 				? `skill-${publicRuntimeConfig.SLUG}`
 				: ''
 
+		const renderLocation = get(
+			this.props.__NEXT_DATA__,
+			'props.pageProps.initialProps.renderLocation',
+			null
+		)
+
```

Comment:
@gcpantazis this `__NEXT_DATA__` object was the only way I could find to access the `renderLocation` prop that is set on the page. Feels a little dirty but the idea here was so that the `render-location-xxx` class would get added even if the page is rendered server-side.

Reply:
Yeah, you'd want to do it like this:

https://github.com/sprucelabsai/ai-spruce-web/blob/dev/pages/_document.js#L14-L18

This would spread initial props for the page into the _document render's props.